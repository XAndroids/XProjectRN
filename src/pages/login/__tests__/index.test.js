import React from "react";
import Login from '../index';
import {render, shallow} from "react-native-testing-library";

describe("login page test", () => {
    it("login page show test", () => {
        const {getByTestId} = render(<Login/>);
        //展示背景图片
        const imageBackground = getByTestId('loginroot')
        //异常：Compared values have no visual difference. Note that you are testing for equality with the stricter `toBe` matcher using `Object.is`. For deep equality only, use `toEqual` instead.
        //如果Check Object使用toEqual()
        //参考：https://jestjs.io/docs/en/using-matchers
        expect(imageBackground.props.style).toEqual({flex: 1, justifyContent: 'center', alignItems: 'center'})
        expect(imageBackground.props.source.testUri).toBe('../../../img/login_background.png')
        //展示登录图片
        const loginImage = getByTestId('loginimage')
        expect(loginImage.props.source.testUri).toBe('../../../img/login_icon.png')
        //展示登录按钮
        const loginButton = getByTestId('loginbutton')
        expect(loginButton.props.style[0]).toEqual({
            width: 250, height: 50, justifyContent: 'center',
            alignItems: 'center'
        })
        expect(loginButton.props.style[1]).toEqual({marginTop: 40, backgroundColor: '#01A3AE'})
        const loginText = getByTestId('logintext')
        expect(loginText.props.style).toEqual({color: 'white'})
        expect(loginText.props.children).toBe('LOGIN TO FEEDLY')

        //展示尝试按钮
        const tryButton = getByTestId('trybutton')
        expect(tryButton.props.style).toEqual([
            {width: 250, height: 50, justifyContent: 'center', alignItems: 'center'},
            {marginTop: 8, backgroundColor: '#AAAAAA'}
        ])
        const tryText = getByTestId('trytext')
        expect(tryText.props.style).toEqual({color: 'white'})
        expect(tryText.props.children).toBe('TRY THINGS OUT')
        //FIXME 不是外部传入的事件如何测试
    })
})

describe("login page Snapshot test", () => {
    it("login page Snapshot test", () => {
        const {toJSON} = render(<Login/>)
        expect(toJSON()).toMatchSnapshot()
    })
})

describe("login page shallow test", () => {
    it("login page shallow test", () => {
        const {output} = shallow(<Login/>)
        expect(output).toMatchSnapshot()
    })
})
