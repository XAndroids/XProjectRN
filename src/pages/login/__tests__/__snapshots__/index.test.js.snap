// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page Snapshot test login page Snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../img/login_background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="loginroot"
  />
  <Image
    source={
      Object {
        "testUri": "../../../img/login_icon.png",
      }
    }
    testID="loginimage"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#01A3AE",
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "width": 250,
      }
    }
    testID="loginbutton"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
      testID="logintext"
    >
      LOGIN TO FEEDLY
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#AAAAAA",
        "height": 50,
        "justifyContent": "center",
        "marginTop": 8,
        "opacity": 1,
        "width": 250,
      }
    }
    testID="trybutton"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
      testID="trytext"
    >
      TRY THINGS OUT
    </Text>
  </View>
</View>
`;

exports[`login page shallow test login page shallow test 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../img/login_background.png",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="loginroot"
>
  <Image
    source={
      Object {
        "testUri": "../../../img/login_icon.png",
      }
    }
    testID="loginimage"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "width": 250,
        },
        Object {
          "backgroundColor": "#01A3AE",
          "marginTop": 40,
        },
      ]
    }
    testID="loginbutton"
  >
    <Component
      style={
        Object {
          "color": "white",
        }
      }
      testID="logintext"
    >
      LOGIN TO FEEDLY
    </Component>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "width": 250,
        },
        Object {
          "backgroundColor": "#AAAAAA",
          "marginTop": 8,
        },
      ]
    }
    testID="trybutton"
  >
    <Component
      style={
        Object {
          "color": "white",
        }
      }
      testID="trytext"
    >
      TRY THINGS OUT
    </Component>
  </TouchableOpacity>
</ImageBackground>
`;
